google/cloud/aiplatform/datasets/image_dataset.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Optional, Sequence, Dict, Tuple
+
+from google.auth import credentials as auth_credentials
+
+from google.cloud.aiplatform import schema
+from google.cloud.aiplatform import utils
+from google.cloud.aiplatform.datasets import (
+    Dataset,
+    NonTabularDatasource,
+    NonTabularDatasourceImportable,
+)
+
+
+class ImageDataset(Dataset):
+    """Managed image dataset resource for AI Platform"""
+
+    _support_metadata_schema_uris = (schema.dataset.metadata.image,)
+
+    _support_import_schema_classes = ("image",)
+
+    def __init__(
+        self,
+        dataset_name: str,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Retrieves an existing managed image dataset given a dataset name or ID.
+
+        Args:
+            dataset_name: (str)
+                Required. A fully-qualified dataset resource name or dataset ID.
+                Example: "projects/123/locations/us-central1/datasets/456" or
+                "456" when project and location are initialized or passed.
+            project: (str) = None
+                Optional project to retrieve dataset from. If not set, project
+                set in aiplatform.init will be used.
+            location: (str) = None
+                Optional location to retrieve dataset from. If not set, location
+                set in aiplatform.init will be used.
+            credentials: Optional[auth_credentials.Credentials] = None,
+                Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+        """
+        super().__init__(
+            dataset_name=dataset_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+    @classmethod
+    def create(
```

Comment:
This is a little strange, I thought the point of the subclass was to only expose the relevant parameters and exclude non-relevant ones (for example, bq_source in the case of image_dataset).

Reply:
Hence, in the branch I provided, the signature was:

```
@classmethod
    def create(
        cls,
        display_name: str,
        gcs_source_uris: Sequence[str],
        import_schema_uri: str,
        data_items_labels: Optional[Dict] = None,
        metadata: Sequence[Tuple[str, str]] = (),
        labels: Optional[Dict] = None,
        project: Optional[str] = None,
        location: Optional[str] = None,
        credentials: Optional[auth_credentials.Credentials] = None,
        sync=True,
    ) -> "Dataset":
```

Reply:
I wonder if it makes more sense to use this instead:

```
    def create(
        cls,
        display_name: str,
        datasource: Union[NonTabularDatasource, NonTabularDatasourceImportable],
        metadata: Sequence[Tuple[str, str]] = (),
        labels: Optional[Dict] = None,
        project: Optional[str] = None,
        location: Optional[str] = None,
        credentials: Optional[auth_credentials.Credentials] = None,
        sync=True,
    ) -> "Dataset":
```

This would automatically give you validation such as:
- dataset_items_labels is only relevant when import_schema_uri is provided
- gcs_source_uris is only relevant when import_schema_uri is provided

It'd also make subclasses even more trivial to write.

Reply:
Probably need @sasha-gitg's opinion on this

Reply:
@ivanmkc to your first feedback,  bq_source was left by mistake, will fix it.

Reply:
Input signature LGTM.